hospital/src/main/java/by/emel/anton/model/dao/implementation/filedao/FileServiceDAO.java

```diff
@@ -34,45 +35,55 @@ public boolean isIdCorrect(String data, int id) {
         return id == idData;
     }
 
-    public int getNextLineId(Path filePath) throws UserDAOException {
+    public int getNextLineId(Path filePath) {
 
         try {
             List<String> fileData = Files.readAllLines(filePath);
             String lineMaxID = fileData.stream().max(new LineIdComaparator()).orElse("0/");
 
             return Integer.parseInt(lineMaxID.split(Constants.SEPARATOR)[0]) + 1;
+
         } catch (IOException e) {
-            throw new UserDAOException("ERROR getNextLineID");
+            throw new UserDaoUncheckedException("ERROR getNextLineID");
         }
     }
 
-    public Map<Integer, List<Integer>> getDoctorPatientsIdMap() throws UserDAOException {
+    public Map<Integer, List<Integer>> getDoctorPatientsIdMap() {
         try {
             return Files.readAllLines(Paths.get(Constants.FILE_PATH_PATIENTS))
                     .stream()
                     .collect(Collectors.groupingBy(line -> Integer.valueOf(line.split(Constants.SEPARATOR)[1]),
                             Collectors.mapping(line -> Integer.valueOf(line.split(Constants.SEPARATOR)[0]), Collectors.toList())));
 
         } catch (IOException e) {
-            throw new UserDAOException("ERROR get Patient id list");
+            throw new UserDaoUncheckedException("ERROR get Patient id list");
         }
     }
 
-    public void addTherapiesToPatient(Patient patient) throws IOException {
-        List<String> listLineThe = Files.readAllLines(Paths.get(Constants.FILE_PATH_THERAPIES)).stream()
-                .filter(l -> Integer.parseInt(l.split(Constants.SEPARATOR)[4]) == patient.getId())
-                .collect(Collectors.toList());
+    public void addTherapiesToPatient(Patient patient) {
 
-        List<Therapy> therapies = listLineThe.stream()
-                .map(this::createTherapyFromLine)
-                .collect(Collectors.toList());
-        therapies.forEach(t -> t.setPatient(patient));
+        try {
+            List<String> listLineThe = Files
+                    .readAllLines(Paths.get(Constants.FILE_PATH_THERAPIES))
+                    .stream()
+                    .filter(line -> Integer.parseInt(line.split(Constants.SEPARATOR)[4]) == patient.getId())
+                    .collect(Collectors.toList());
 
-        patient.setTherapies(therapies);
+            List<Therapy> therapies = listLineThe
+                    .stream()
+                    .map(this::createTherapyFromLine)
+                    .collect(Collectors.toList());
 
+            therapies.forEach(t -> t.setPatient(patient));
+
+            patient.setTherapies(therapies);
+
+        } catch (IOException e) {
+            throw new TherapyDaoUncheckedException();
+        }
     }
 
-    public void addPatientsToDoctor(Doctor doctor) throws UserDAOException {
+    public void addPatientsToDoctor(Doctor doctor) throws UserDaoUncheckedException {
```

Comment:
Remove unchecked exception from `throws` declaration. Do not use work `unchecked` in the class name

Reply:
done